src/view/gui/fMainWindow.cpp

```diff
@@ -9662,7 +9689,7 @@ void fMainWindow::closeEvent(QCloseEvent* event)
   }
 }
 
-void fMainWindow::updateProgress(int progress, std::string message, int max)
+void fMainWindow::updateProgress(qint64 progress, std::string message, qint64 max)
```

Comment:
Will this break other applications that connect to the updateProgress slot? Or does it automatically cast?

Reply:
I guess it should auto cast, but we should test. There are no connect failure errors (due to mismatched signals/slots) thrown which are usually thrown at compile time.

Reply:
Good catch. @ashishsingh18: There is actually no reason to use special data types unless absolutely needed, right?

Reply:
From [Qt doc](https://doc.qt.io/qt-5/qtglobal.html#qint64-typedef):

```
typedef qint64
Typedef for long long int (__int64 on Windows). This type is guaranteed to be 64-bit on all platforms supported by Qt.
```

Shouldn't create any issue, just that there is no reason to use it...

Reply:
I don't think I understood the 'no reason to use it' part.
The progress we are getting from Qt Network class is in qint64 and that is getting propagated to our GUI. We can replace qint64 with long long in our code for easy reading. But I don't think typedef is a problem. I guess @AlexanderGetka-cbica's quesion was - since there is a mismatch between qint64( long long int) and int( which is what our older UpdateProgress slot was using), will this be a problem with other signals that connect to the same slot. Or would we need to update all of them?

Reply:
I guess this is a stylistic thing but I prefer not to use "custom" data-types. 🤷 

As far as @AlexanderGetka-cbica's question goes, the casting from `int` to `long long` shouldn't be an issue for other applications.